fastly/block_fastly_service_v1_logging_kinesis.go

```diff
@@ -254,17 +263,24 @@ func (h *KinesisServiceAttributeHandler) Register(s *schema.Resource) error {
 
 		"access_key": {
 			Type:        schema.TypeString,
-			Required:    true,
+			Optional:    true,
 			Sensitive:   true,
 			Description: "The AWS access key to be used to write to the stream",
 		},
 
 		"secret_key": {
 			Type:        schema.TypeString,
-			Required:    true,
+			Optional:    true,
 			Sensitive:   true,
 			Description: "The AWS secret access key to authenticate with",
 		},
+
+		"iam_role": {
+			Type:        schema.TypeString,
+			Optional:    true,
+			Description: "The Amazon Resource Name (ARN) for the IAM role granting Fastly access to Kinesis. Not required if `access_key` and `secret_key` are provided.",
+			Sensitive:   false,
+		},
```

Comment:
Might be worth using [`ConflictsWith`](https://github.com/hashicorp/terraform-plugin-sdk/blob/v2.5.0/helper/schema/schema.go#L196) to ensure the user can't accidentally set both Access/Secret keys _and_ an IAM role as we know from the CLI implementation that they are mutually exclusive. Then you can add a test to validate the behaviour.

Reply:
Cool, I did not know about `ConflictsWith`. Thanks for sharing that.

Reply:
I tried this out, but `ConflictsWith` does not appear to work in this context. I get the following errors:
```
    block_fastly_service_v1_s3logging_test.go:233: Step 1/1 error: Error running pre-apply refresh: exit status 1
        
        Error: Invalid provider configuration
        
        InternalValidate: Internal validation of the provider failed! This is always
        a bug
        with the provider itself, and not a user issue. Please report
        this bug:
        
        2 errors occurred:
        	* resource fastly_service_compute: ConflictsWith: s3_secret_key
        configuration block reference (s3logging.s3_iam_role) can only be used with
        TypeList and MaxItems: 1 configuration blocks
        	* resource fastly_service_v1: ConflictsWith: s3_iam_role configuration block
        reference (s3logging.s3_access_key) can only be used with TypeList and
        MaxItems: 1 configuration blocks
        
        
        
        Provider "registry.terraform.io/hashicorp/fastly" requires explicit
        configuration. Add a provider block to the root module and configure the
        provider's required arguments as described in the provider documentation.
        
--- FAIL: TestAccFastlyServiceV1_s3logging_basic_compute (14.38s)
=== CONT  TestAccFastlyServiceV1_s3logging_basic
    block_fastly_service_v1_s3logging_test.go:181: Step 1/2 error: Error running pre-apply refresh: exit status 1
        
        Error: Invalid provider configuration
        
        InternalValidate: Internal validation of the provider failed! This is always
        a bug
        with the provider itself, and not a user issue. Please report
        this bug:
        
        2 errors occurred:
        	* resource fastly_service_compute: ConflictsWith: s3_secret_key
        configuration block reference (s3logging.s3_iam_role) can only be used with
        TypeList and MaxItems: 1 configuration blocks
        	* resource fastly_service_v1: ConflictsWith: s3_access_key configuration
        block reference (s3logging.s3_iam_role) can only be used with TypeList and
        MaxItems: 1 configuration blocks
```

A lot of the `s3logging` logic depends on the type being a set so I'm not sure about changing that and even so I don't think we want to have the `MaxItems` constraint in this case.

Maybe it's better just to rely on the server error to enforce this:
```
    block_fastly_service_v1_s3logging_test.go:181: Step 1/2 error: Error running apply: exit status 1
        
        Error: 400 - Bad Request:
        
            Title:  'iam_role' can't be used when 'access_key' or 'secret_key' are provided
        
          on terraform_plugin_test.tf line 2, in resource "fastly_service_v1" "foo":
           2: resource "fastly_service_v1" "foo" {
```

Reply:
Ahhh, the 'ole TypeSet issue keeps coming back to bite us. Sure let's revert the ConflictsWith attribute 👍🏻

Reply:
Sounds good. I didn't actually push up those changes so there's nothing to revert on this branch.